PR: https://github.com/odoo/odoo/pull/

From: 0f7930353d8fd283380c7e50658f2656ebdd7845
From: dbh
Date: 2018-01-04 12:26:52

Structural Changes: 12
Total Changes: 24

[IMP] base,base_address_city,base_setup: give warning if fields have same label to easily identify duplicate field string.

================================= pseudo patch: =================================

--- a/odoo/addons/base/models/ir_actions.py
+++ b/odoo/addons/base/models/ir_actions.py
@@ -173,7 +173,7 @@ class IrActionsActWindow(models.Model):
                                  help="View type: Tree type to use for the tree view, set to 'tree' for a hierarchical tree view, or 'form' for a regular list view")
     usage = fields.Char(string='Action Usage',
                         help="Used to filter menu and home actions from the user form.")
-    view_ids = fields.One2many('ir.actions.act_window.view', 'act_window_id', string='Views')
+    view_ids = fields.One2many('ir.actions.act_window.view', 'act_window_id', string='No of Views')
     views = fields.Binary(compute='_compute_views',
                           help="This function field computes the ordered list of views that should be enabled " \
                                "when displaying the result of an action, federating view mode, views and " \
@@ -358,7 +358,7 @@ class IrActionsServer(models.Model):
                                    "based on the sequence. Low number means high priority.")
     model_id = fields.Many2one('ir.model', string='Model', required=True, ondelete='cascade',
                                help="Model on which the server action runs.")
-    model_name = fields.Char(related='model_id.model', readonly=True, store=True)
+    model_name = fields.Char(related='model_id.model', string='Model Name', readonly=True, store=True)
     # Python code
     code = fields.Text(string='Python Code', groups='base.group_system',
                        default=DEFAULT_PYTHON_CODE,

--- a/odoo/addons/base/models/ir_sequence.py
+++ b/odoo/addons/base/models/ir_sequence.py
@@ -126,7 +126,7 @@ class IrSequence(models.Model):
     suffix = fields.Char(help="Suffix value of the record for the sequence", trim=False)
     number_next = fields.Integer(string='Next Number', required=True, default=1, help="Next number of this sequence")
     number_next_actual = fields.Integer(compute='_get_number_next_actual', inverse='_set_number_next_actual',
-                                        string='Next Number',
+                                        string='Actual Next Number',
                                         help="Next number that will be used. This number can be incremented "
                                         "frequently so the displayed value might already be obsolete")
     number_increment = fields.Integer(string='Step', required=True, default=1,
@@ -340,7 +340,7 @@ class IrSequenceDateRange(models.Model):
     sequence_id = fields.Many2one("ir.sequence", string='Main Sequence', required=True, ondelete='cascade')
     number_next = fields.Integer(string='Next Number', required=True, default=1, help="Next number of this sequence")
     number_next_actual = fields.Integer(compute='_get_number_next_actual', inverse='_set_number_next_actual',
-                                        string='Next Number',
+                                        string='Actual Next Number',
                                         help="Next number that will be used. This number can be incremented "
                                              "frequently so the displayed value might already be obsolete")
 

--- a/odoo/addons/base/models/ir_ui_view.py
+++ b/odoo/addons/base/models/ir_ui_view.py
@@ -195,7 +195,7 @@ class View(models.Model):
                              ('search', 'Search'),
                              ('qweb', 'QWeb')], string='View Type')
     arch = fields.Text(compute='_compute_arch', inverse='_inverse_arch', string='View Architecture', nodrop=True)
-    arch_base = fields.Text(compute='_compute_arch_base', inverse='_inverse_arch_base', string='View Architecture')
+    arch_base = fields.Text(compute='_compute_arch_base', inverse='_inverse_arch_base', string='Base View Architecture')
     arch_db = fields.Text(string='Arch Blob', translate=xml_translate, oldname='arch')
     arch_fs = fields.Char(string='Arch Filename')
     inherit_id = fields.Many2one('ir.ui.view', string='Inherited View', ondelete='restrict', index=True)
@@ -207,7 +207,7 @@ class View(models.Model):
                          help="ID of the view defined in xml file")
     groups_id = fields.Many2many('res.groups', 'ir_ui_view_group_rel', 'view_id', 'group_id',
                                  string='Groups', help="If this field is empty, the view applies to all users. Otherwise, the view applies to the users of those groups only.")
-    model_ids = fields.One2many('ir.model.data', 'res_id', domain=[('model', '=', 'ir.ui.view')], auto_join=True)
+    model_ids = fields.One2many('ir.model.data', 'res_id', string="Models", domain=[('model', '=', 'ir.ui.view')], auto_join=True)
     create_date = fields.Datetime(readonly=True)
     write_date = fields.Datetime(string='Last Modification Date', readonly=True)
 

--- a/odoo/addons/base/models/res_partner.py
+++ b/odoo/addons/base/models/res_partner.py
@@ -213,7 +213,7 @@ class Partner(models.Model):
     # technical field used for managing commercial fields
     commercial_partner_id = fields.Many2one('res.partner', compute='_compute_commercial_partner',
                                              string='Commercial Entity', store=True, index=True)
-    commercial_partner_country_id = fields.Many2one('res.country', related='commercial_partner_id.country_id', store=True)
+    commercial_partner_country_id = fields.Many2one('res.country', related='commercial_partner_id.country_id', string='Commercial Entity Country', store=True)
     commercial_company_name = fields.Char('Company Name Entity', compute='_compute_commercial_company_name',
                                           store=True)
     company_name = fields.Char('Company Name')

--- a/odoo/addons/base/views/ir_sequence_views.xml
+++ b/odoo/addons/base/views/ir_sequence_views.xml
@@ -28,14 +28,14 @@
                           <group>
                             <field name="padding"/>
                             <field name="number_increment"/>
-                            <field name="number_next_actual" attrs="{'invisible': [('use_date_range', '=', True)]}"/>
+                            <field name="number_next_actual" string="Next Number" attrs="{'invisible': [('use_date_range', '=', True)]}"/>
                           </group>
                         </group>
                         <field name="date_range_ids" attrs="{'invisible': [('use_date_range', '=', False)]}">
                             <tree string="Sequences" editable="top">
                                 <field name="date_from"/>
                                 <field name="date_to"/>
-                                <field name="number_next_actual"/>
+                                <field name="number_next_actual" string="Next Number"/>
                             </tree>
                         </field>
                         <group col="3" string="Legend (for prefix, suffix)">
@@ -78,7 +78,7 @@
                     <field name="prefix"/>
                     <field name="padding"/>
                     <field name="company_id" groups="base.group_multi_company"/>
-                    <field name="number_next_actual"/>
+                    <field name="number_next_actual" string="Next Number"/>
                     <field name="number_increment"/>
                     <field name="implementation"/>
                 </tree>

--- a/odoo/addons/base/views/ir_ui_view_views.xml
+++ b/odoo/addons/base/views/ir_ui_view_views.xml
@@ -26,7 +26,7 @@
                         <page string="Architecture">
                             <button type="object" name="open_translations"
                                 string="Edit Translations" class="oe_link oe_right"/>
-                            <field name="arch_base" widget="ace" options="{'mode': 'xml'}"/>
+                            <field name="arch_base" string="View Architecture" widget="ace" options="{'mode': 'xml'}"/>
                         </page>
                         <page string="Access Rights">
                             <field name="groups_id"/>

--- a/odoo/addons/base/wizard/base_module_uninstall_views.xml
+++ b/odoo/addons/base/wizard/base_module_uninstall_views.xml
@@ -34,7 +34,7 @@
                         </kanban>
                     </field>
                     <h3>The following documents will be permanently lost</h3>
-                    <field name="model_ids">
+                    <field name="model_ids" string="Models">
                         <tree string="Models">
                             <field name="name" string="Document"/>
                             <field name="count"/>
